andela/ah-frontend-prime
# #164798254 Social-authentication using google and facebook

**What does this PR do?**
    Implements functionality of users being able to login using facebook and google

**Description of tasks to be completed**

- install react-facebook-login and react-google-login modules

- create a social authentication component using redux

- Access the backend API's to log in a user into the system

**How should this be manually tested?**
   -  Visit the login page
   - Click on either google or facebook to log into the system using your social media account
   - If successful, a toast will appear informing the user of their successful authentication and login into 
      the system
   - User will be navigated to the home page

**Any background context**

**Screenshots**

   - Google Authentication
   
   - On the  login page, click on the **LOGIN WITH GOOGLE** button:

<img width="1439" alt="Screen Shot 2019-05-12 at 22 59 16" src="https://user-images.githubusercontent.com/26630001/57587205-0f665c00-750a-11e9-94ff-4c56b71be583.png">

   - Choose an account from the Google accounts dialog:

<img width="488" alt="Screen Shot 2019-05-09 at 17 02 05" src="https://user-images.githubusercontent.com/26630001/57459640-49bac980-727c-11e9-8c26-bc3b937d66ed.png">

  - If successful, you are logged in and redirected to the home page:

<img width="1418" alt="Screen Shot 2019-05-12 at 22 59 44" src="https://user-images.githubusercontent.com/26630001/57587208-1beab480-750a-11e9-9c00-07b4e92367a3.png">


     
**What are the relevant pivotal tracker stories**
    [#164798254](https://www.pivotaltracker.com/story/show/164798254)
